### ISSUE TITLE
Startup Crash in v18 RC1 

### ISSUE THREAD

**Issue by TiagoJSilva** [xbmc/xbmc#14937]
### Describe the bug
Here is a clear and concise description of what the problem is:

When starting Kodi 18 RC1 in both x32 and x64 builds, on both a current installation and a portable version, Kodi does not start and hangs on a black screen. Sometimes, the dialog "Kodi has stopped responding" comes up over the black screen; other times, Ctrl-Alt-Del is needed to invoke Task Manager and then the dialog is visible. The only solution is to kill Kodi. RC1 is unusable in a system where all v18 beta builds have been usable.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Well, I'd expect Kodi v18 to start, like in the betas...

## Actual Behavior

But it always crashes.

### To Reproduce

1. I don't know how to reproduce the bug on other systems, don't have another system or even variations in hardware to test, this is an integrated HTPC.
2. Find an Nvidia ION-based HTPC running Windows 7 x64 and install v18 RC1?
3. Start Kodi

### Debuglog

The debuglog can be found here:
https://paste.kodi.tv/cawigujako.kodi (x86)
https://paste.kodi.tv/ehejatuvep.kodi (x64)
They hang on the same place, but the stack traces are different.
In x86 the stack trace is
```
# 0 RaiseException
# 1 process_jpeg
# 2 cp_run_plugins_step
# 3 process_jpeg
# 4 process_jpeg
# 5
# 6
# 7
# 8 cp_run_plugins_step
# 9 BaseThreadInitThunk
#10 RtlInitializeExceptionChain
#11 RtlInitializeExceptionChain
```
In x64 the stack trace is empty.

There are no screenshots to be taken (blank screen)

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] Linux
 - [ ] OSX
 - [ ] Raspberri-Pi
 - [x] Windows
 - [ ] Windows UWP

 - Operating system version/name: Windows 7 x64
 - Nvidia driver 342.01 WHQL (latest and final version for this chipset)
 - Kodi version: 18.0 RC1
 - Hardware: Asus S1-AT5NM10E (Atom D525, Nvidia ION-II GT218) [(full specifications)](https://www.asus.com/pt/Mini-PCs/S1AT5NM10E/specifications/)

Not sure if relevant, but x86 stack trace indicates there can be a relation because it mentions plugins:
I noticed that the RC1 Kodi installer, unlike all installers beforehand, did not show the option to have InputStream plug-ins as an optional install. The only item in the optional installs section of the RC1 installer was the VC Redistributable. 

Kodi Crashlog DMPs:
x86: [kodi_crashlog-18.0-RC1 Git_20181121-314c302559-20181124-153102.zip](https://github.com/xbmc/xbmc/files/2613061/kodi_crashlog-18.0-RC1.Git_20181121-314c302559-20181124-153102.zip)
x64: [kodi_crashlog-18.0-RC1 Git_20181121-314c302559-20181124-153117.zip](https://github.com/xbmc/xbmc/files/2613062/kodi_crashlog-18.0-RC1.Git_20181121-314c302559-20181124-153117.zip)

Note: this issue report duplicates this [Kodi forum thread](https://forum.kodi.tv/showthread.php?tid=337774)

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by TiagoJSilva** [comment]
I was advised to test all the nightlies between Beta5 and RC1, and found that the first built that exhibits the issue is KodiSetup-20181117-f01060fc-master-x64.exe
[kodi.log](https://paste.kodi.tv/aguteduwaz.kodi) ends with the same messages as RC1

[kodi_crashlog-18.0-RC1 Git_20181117-f01060fc8c-20181126-021810.zip](https://github.com/xbmc/xbmc/files/2613877/kodi_crashlog-18.0-RC1.Git_20181117-f01060fc8c-20181126-021810.zip)

**Comment by TiagoJSilva** [comment]
Ooops, didn't mean to close, it's not solved... Must've clicked "Close and comment" (and was about to do it again...)

**Comment by TiagoJSilva** [comment]
Another poster on the Kodi forums [was having startup issues with Leia RC1](https://forum.kodi.tv/showthread.php?tid=337937&pid=2796907#pid2796907), but on an Intel chipset. I tested his "solution", which was replacing the d3compiler47.dll file and it also "solves" my issue (at least in starting up Kodi v18 RC1: other bugs may appear later due to .dll mismatch)

- Installed latest nightly
- Confirmed it still crashes
- Overwrote the d3dcompiler47.dll of 20181128 with the one from 20181116
- Leia RC1 starts!

The original poster has his kodi.log on that forum thread.

**Comment by TiagoJSilva** [comment]
Just tested RC2, it still hangs on startup and the "fix" also works

**Comment by pkerling** [comment]
@Rechi @afedchin Do you know of recent changes to d3dcompiler47.dll?

**Comment by afedchin** [comment]
Wun SDK was updated, it means that may be vc2017 redistributable installed required

**Comment by TiagoJSilva** [comment]
Where do you prefer this conversation to continue? On the forums or here?

afedchin asked for this information on the forum, I replied here because here I can just upload the screenshots of the working and non working d3dcompiler.dll without having to upload them to something like imgur first.

![crashing d3dcompiler47.dll](https://user-images.githubusercontent.com/4505161/50317692-ca874d80-04b4-11e9-9d3e-3a6804824c8a.PNG)
![working d3dcompiler47.dll](https://user-images.githubusercontent.com/4505161/50317693-ca874d80-04b4-11e9-98f8-2a7bd94f91fb.PNG)

**Comment by TiagoJSilva** [comment]
I'll try to do it tonight. After Saturday, I won't be able to provide more data until the 26th or 27th, because of the holidays.

**Comment by TiagoJSilva** [comment]
Well, how about that? Installing vcredist manually just solved it!

Downloaded vcredist.x64.exe from https://aka.ms/vs/15/release/vc_redist.x64.exe as directed on the forum thread. Properties/Details of the .exe indicate it's version 14.16.27012.6.
Installed VC 2017 Redist from the .exe and restarted.
Went back to my previous RC2 install that did not open.
Confirmed d3dcompiler_47.dll is version 17763
Started Kodi RC2 in Portable Mode
--> It's alive!
Uninstalled my current "daily driver" v18 Beta4 installation.
Installed a fresh v18 RC2 to load my current user library
Confirmed d3dcompiler_47.dll is the "bad" version (17763)
---> it also works!

RC2 is now working. What changed by installing vcredist manually?

These are all the vcredists I have installed, according to Control Panel:
![all vcredist](https://user-images.githubusercontent.com/4505161/50369993-8a989700-0596-11e9-866e-5153afa795dc.PNG)
